pkg/common/utils_test.go

```diff
@@ -74,3 +76,31 @@ func TestRemoveItem(t *testing.T) {
 		}
 	}
 }
+
+// Test ParseResponseForKeywords with a reader that contains the keyword and a reader that doesn't.
+func TestParseResponseForKeywords(t *testing.T) {
```

Comment:
suggestion: Since we have multiple test cases with the exact same setup we might benefit from a table-driven test. We could also maybe throw in some edge cases?

https://google.github.io/styleguide/go/best-practices#tests
```
If inputs are similar, consider unifying them into a table-driven test while keeping the logic inlined in the loop. This helps to avoid repetition while keeping the validation and failure in the Test.
```

Ex:
```
func TestParseResponseForKeywords(t *testing.T) {
	testCases := []struct {
		name     string
		input    string
		keywords []string
		expected bool
	}{
		{
			name:     "Should find keyword",
			input:    "ey: abc",
			keywords: []string{"ey"},
			expected: true,
		},
		{
			name:     "Should not find keyword",
			input:    "fake response",
			keywords: []string{"ey"},
			expected: false,
		},
		{
			name:     "Empty string",
			input:    "",
			keywords: []string{"ey"},
			expected: false,
		},
		{
			name:     "Keyword at end",
			input:    "abc ey",
			keywords: []string{"ey"},
			expected: true,
		},
		{
			name:     "Keyword at start",
			input:    "ey abc",
			keywords: []string{"ey"},
			expected: true,
		},
	}

	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			testReader := strings.NewReader(tc.input)
			testReadCloser := io.NopCloser(testReader)
			found, err := ParseResponseForKeywords(testReadCloser, tc.keywords)

			if err != nil {
				t.Errorf("Error: %v", err)
			}

			if found != tc.expected {
				t.Errorf("Expected %v, got %v", tc.expected, found)
			}
		})
	}
}
```

Reply:
ah true